parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -414,4 +414,16 @@ write_files:
   encoding: gzip
   owner: root
   content: !!binary |
-    {{GetVariableProperty "cloudInitData" "customSearchDomainsScript"}}
\ No newline at end of file
+    {{GetVariableProperty "cloudInitData" "customSearchDomainsScript"}}
+
+{{ if EnableBootstrapProfileContainerRegistry -}}
+# TODO(binxi): need to update for sovereign cloud.
+- path: /etc/containerd/certs.d/mcr.microsoft.com/hosts.toml
+  permissions: "0644"
+  owner: root
+  content: |
+    server = "https://mcr.microsoft.com"
```

Comment:
>server [](http://example.com/codeflow?start=4&length=6)

don't need this field actually? it's only for fallback. suggest to remove

[https://github.com/containerd/containerd/blob/main/docs/hosts.md#server-field](https://github.com/containerd/containerd/blob/main/docs/hosts.md#server-field)

Reply:
```
server specifies the default server for this registry host namespace.

When host(s) are specified, the hosts will be tried first in the order listed. If all host(s) are tried then server will be used as a fallback.

If server is not specified then the image's registry host namespace will automatically be used.
```

Do you mean, if the server and registry host namespace are same, then we don't need the server field?

Reply:
removed "server" field since the server and registry host namespace are same and if server is not specified then the image's registry host namespace will automatically be used.